// couchbase/dist/binding.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CppTxnExternalException = exports.CppTxnFailureType = exports.CppScanSort = exports.CppReplicateTo = exports.CppPersistTo = exports.CppStoreSemantics = exports.CppImplSubdocOpcode = exports.CppKeyValueStatusCode = exports.CppErrcNetwork = exports.CppErrcFieldLevelEncryption = exports.CppErrcManagement = exports.CppErrcView = exports.CppErrcSearch = exports.CppErrcAnalytics = exports.CppErrcQuery = exports.CppErrcKeyValue = exports.CppErrcCommon = exports.CppDurabilityLevel = exports.CppAnalyticsResponseAnalyticsStatus = exports.CppViewSortOrder = exports.CppViewScanConsistency = exports.CppViewOnError = exports.CppServiceType = exports.CppSearchScanConsistency = exports.CppSearchHighlightStyle = exports.CppQueryScanConsistency = exports.CppQueryProfile = exports.CppDiagPingState = exports.CppDiagEndpointState = exports.CppDiagClusterState = exports.CppDesignDocumentNamespace = exports.CppAnalyticsScanConsistency = exports.CppProtocolSubdocOpcode = exports.CppRetryReason = exports.CppManagementRbacAuthDomain = exports.CppManagementEventingFunctionProcessingStatus = exports.CppManagementEventingFunctionDeploymentStatus = exports.CppManagementEventingFunctionStatus = exports.CppManagementEventingFunctionBucketAccess = exports.CppManagementEventingFunctionLogLevel = exports.CppManagementEventingFunctionLanguageCompatibility = exports.CppManagementEventingFunctionDcpBoundary = exports.CppManagementClusterBucketStorageBackend = exports.CppManagementClusterBucketConflictResolution = exports.CppManagementClusterBucketEvictionPolicy = exports.CppManagementClusterBucketCompression = exports.CppManagementClusterBucketType = exports.CppManagementAnalyticsCouchbaseLinkEncryptionLevel = exports.zeroCas = void 0;
/*
eslint
jsdoc/require-jsdoc: off,
@typescript-eslint/no-empty-interface: off
*/
const bindings_1 = __importDefault(require("bindings"));
exports.zeroCas = 0;
//#region Autogenerated Bindings
var CppManagementAnalyticsCouchbaseLinkEncryptionLevel;
(function (CppManagementAnalyticsCouchbaseLinkEncryptionLevel) {
})(CppManagementAnalyticsCouchbaseLinkEncryptionLevel = exports.CppManagementAnalyticsCouchbaseLinkEncryptionLevel || (exports.CppManagementAnalyticsCouchbaseLinkEncryptionLevel = {}));
var CppManagementClusterBucketType;
(function (CppManagementClusterBucketType) {
})(CppManagementClusterBucketType = exports.CppManagementClusterBucketType || (exports.CppManagementClusterBucketType = {}));
var CppManagementClusterBucketCompression;
(function (CppManagementClusterBucketCompression) {
})(CppManagementClusterBucketCompression = exports.CppManagementClusterBucketCompression || (exports.CppManagementClusterBucketCompression = {}));
var CppManagementClusterBucketEvictionPolicy;
(function (CppManagementClusterBucketEvictionPolicy) {
})(CppManagementClusterBucketEvictionPolicy = exports.CppManagementClusterBucketEvictionPolicy || (exports.CppManagementClusterBucketEvictionPolicy = {}));
var CppManagementClusterBucketConflictResolution;
(function (CppManagementClusterBucketConflictResolution) {
})(CppManagementClusterBucketConflictResolution = exports.CppManagementClusterBucketConflictResolution || (exports.CppManagementClusterBucketConflictResolution = {}));
var CppManagementClusterBucketStorageBackend;
(function (CppManagementClusterBucketStorageBackend) {
})(CppManagementClusterBucketStorageBackend = exports.CppManagementClusterBucketStorageBackend || (exports.CppManagementClusterBucketStorageBackend = {}));
var CppManagementEventingFunctionDcpBoundary;
(function (CppManagementEventingFunctionDcpBoundary) {
})(CppManagementEventingFunctionDcpBoundary = exports.CppManagementEventingFunctionDcpBoundary || (exports.CppManagementEventingFunctionDcpBoundary = {}));
var CppManagementEventingFunctionLanguageCompatibility;
(function (CppManagementEventingFunctionLanguageCompatibility) {
})(CppManagementEventingFunctionLanguageCompatibility = exports.CppManagementEventingFunctionLanguageCompatibility || (exports.CppManagementEventingFunctionLanguageCompatibility = {}));
var CppManagementEventingFunctionLogLevel;
(function (CppManagementEventingFunctionLogLevel) {
})(CppManagementEventingFunctionLogLevel = exports.CppManagementEventingFunctionLogLevel || (exports.CppManagementEventingFunctionLogLevel = {}));
var CppManagementEventingFunctionBucketAccess;
(function (CppManagementEventingFunctionBucketAccess) {
})(CppManagementEventingFunctionBucketAccess = exports.CppManagementEventingFunctionBucketAccess || (exports.CppManagementEventingFunctionBucketAccess = {}));
var CppManagementEventingFunctionStatus;
(function (CppManagementEventingFunctionStatus) {
})(CppManagementEventingFunctionStatus = exports.CppManagementEventingFunctionStatus || (exports.CppManagementEventingFunctionStatus = {}));
var CppManagementEventingFunctionDeploymentStatus;
(function (CppManagementEventingFunctionDeploymentStatus) {
})(CppManagementEventingFunctionDeploymentStatus = exports.CppManagementEventingFunctionDeploymentStatus || (exports.CppManagementEventingFunctionDeploymentStatus = {}));
var CppManagementEventingFunctionProcessingStatus;
(function (CppManagementEventingFunctionProcessingStatus) {
})(CppManagementEventingFunctionProcessingStatus = exports.CppManagementEventingFunctionProcessingStatus || (exports.CppManagementEventingFunctionProcessingStatus = {}));
var CppManagementRbacAuthDomain;
(function (CppManagementRbacAuthDomain) {
})(CppManagementRbacAuthDomain = exports.CppManagementRbacAuthDomain || (exports.CppManagementRbacAuthDomain = {}));
var CppRetryReason;
(function (CppRetryReason) {
})(CppRetryReason = exports.CppRetryReason || (exports.CppRetryReason = {}));
var CppProtocolSubdocOpcode;
(function (CppProtocolSubdocOpcode) {
})(CppProtocolSubdocOpcode = exports.CppProtocolSubdocOpcode || (exports.CppProtocolSubdocOpcode = {}));
var CppAnalyticsScanConsistency;
(function (CppAnalyticsScanConsistency) {
})(CppAnalyticsScanConsistency = exports.CppAnalyticsScanConsistency || (exports.CppAnalyticsScanConsistency = {}));
var CppDesignDocumentNamespace;
(function (CppDesignDocumentNamespace) {
})(CppDesignDocumentNamespace = exports.CppDesignDocumentNamespace || (exports.CppDesignDocumentNamespace = {}));
var CppDiagClusterState;
(function (CppDiagClusterState) {
})(CppDiagClusterState = exports.CppDiagClusterState || (exports.CppDiagClusterState = {}));
var CppDiagEndpointState;
(function (CppDiagEndpointState) {
})(CppDiagEndpointState = exports.CppDiagEndpointState || (exports.CppDiagEndpointState = {}));
var CppDiagPingState;
(function (CppDiagPingState) {
})(CppDiagPingState = exports.CppDiagPingState || (exports.CppDiagPingState = {}));
var CppQueryProfile;
(function (CppQueryProfile) {
})(CppQueryProfile = exports.CppQueryProfile || (exports.CppQueryProfile = {}));
var CppQueryScanConsistency;
(function (CppQueryScanConsistency) {
})(CppQueryScanConsistency = exports.CppQueryScanConsistency || (exports.CppQueryScanConsistency = {}));
var CppSearchHighlightStyle;
(function (CppSearchHighlightStyle) {
})(CppSearchHighlightStyle = exports.CppSearchHighlightStyle || (exports.CppSearchHighlightStyle = {}));
var CppSearchScanConsistency;
(function (CppSearchScanConsistency) {
})(CppSearchScanConsistency = exports.CppSearchScanConsistency || (exports.CppSearchScanConsistency = {}));
var CppServiceType;
(function (CppServiceType) {
})(CppServiceType = exports.CppServiceType || (exports.CppServiceType = {}));
var CppViewOnError;
(function (CppViewOnError) {
})(CppViewOnError = exports.CppViewOnError || (exports.CppViewOnError = {}));
var CppViewScanConsistency;
(function (CppViewScanConsistency) {
})(CppViewScanConsistency = exports.CppViewScanConsistency || (exports.CppViewScanConsistency = {}));
var CppViewSortOrder;
(function (CppViewSortOrder) {
})(CppViewSortOrder = exports.CppViewSortOrder || (exports.CppViewSortOrder = {}));
var CppAnalyticsResponseAnalyticsStatus;
(function (CppAnalyticsResponseAnalyticsStatus) {
})(CppAnalyticsResponseAnalyticsStatus = exports.CppAnalyticsResponseAnalyticsStatus || (exports.CppAnalyticsResponseAnalyticsStatus = {}));
var CppDurabilityLevel;
(function (CppDurabilityLevel) {
})(CppDurabilityLevel = exports.CppDurabilityLevel || (exports.CppDurabilityLevel = {}));
var CppErrcCommon;
(function (CppErrcCommon) {
})(CppErrcCommon = exports.CppErrcCommon || (exports.CppErrcCommon = {}));
var CppErrcKeyValue;
(function (CppErrcKeyValue) {
})(CppErrcKeyValue = exports.CppErrcKeyValue || (exports.CppErrcKeyValue = {}));
var CppErrcQuery;
(function (CppErrcQuery) {
})(CppErrcQuery = exports.CppErrcQuery || (exports.CppErrcQuery = {}));
var CppErrcAnalytics;
(function (CppErrcAnalytics) {
})(CppErrcAnalytics = exports.CppErrcAnalytics || (exports.CppErrcAnalytics = {}));
var CppErrcSearch;
(function (CppErrcSearch) {
})(CppErrcSearch = exports.CppErrcSearch || (exports.CppErrcSearch = {}));
var CppErrcView;
(function (CppErrcView) {
})(CppErrcView = exports.CppErrcView || (exports.CppErrcView = {}));
var CppErrcManagement;
(function (CppErrcManagement) {
})(CppErrcManagement = exports.CppErrcManagement || (exports.CppErrcManagement = {}));
var CppErrcFieldLevelEncryption;
(function (CppErrcFieldLevelEncryption) {
})(CppErrcFieldLevelEncryption = exports.CppErrcFieldLevelEncryption || (exports.CppErrcFieldLevelEncryption = {}));
var CppErrcNetwork;
(function (CppErrcNetwork) {
})(CppErrcNetwork = exports.CppErrcNetwork || (exports.CppErrcNetwork = {}));
var CppKeyValueStatusCode;
(function (CppKeyValueStatusCode) {
})(CppKeyValueStatusCode = exports.CppKeyValueStatusCode || (exports.CppKeyValueStatusCode = {}));
var CppImplSubdocOpcode;
(function (CppImplSubdocOpcode) {
})(CppImplSubdocOpcode = exports.CppImplSubdocOpcode || (exports.CppImplSubdocOpcode = {}));
var CppStoreSemantics;
(function (CppStoreSemantics) {
})(CppStoreSemantics = exports.CppStoreSemantics || (exports.CppStoreSemantics = {}));
var CppPersistTo;
(function (CppPersistTo) {
})(CppPersistTo = exports.CppPersistTo || (exports.CppPersistTo = {}));
var CppReplicateTo;
(function (CppReplicateTo) {
})(CppReplicateTo = exports.CppReplicateTo || (exports.CppReplicateTo = {}));
var CppScanSort;
(function (CppScanSort) {
})(CppScanSort = exports.CppScanSort || (exports.CppScanSort = {}));
var CppTxnFailureType;
(function (CppTxnFailureType) {
})(CppTxnFailureType = exports.CppTxnFailureType || (exports.CppTxnFailureType = {}));
var CppTxnExternalException;
(function (CppTxnExternalException) {
})(CppTxnExternalException = exports.CppTxnExternalException || (exports.CppTxnExternalException = {}));
// Load it with require
const binding = (0, bindings_1.default)('couchbase_impl');
exports.default = binding;
